File: proxy/tests/script_engine/url_path.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local url_path = {
	request = function(flow)
		if flow.request.url.path == "/some/path" then
			flow.request.url.path = "/another/path"
		end
	end,
}
Extensions = { url_path }

File: proxy/tests/script_engine/url_path.js

/// <reference path="../../script_libs/js/index.d.ts" />
/** @type {Extension} */
const url_path = {
  request(flow) {
    if (flow.request.url.path == "/some/path") {
      flow.request.url.path = "/another/path";
    }
  }
};
globalThis.extensions = [url_path]
